src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/publicip/sdk/AzurePublicIPAddressOperationSDKTest.java

```diff
@@ -0,0 +1,246 @@
+    }
+
+    // test case: When calling the handleSecurityIssues method, it must verify that
+    // is call was successful.
+    @Test
+    public void testHandleSecurityIssuesSuccessfully() throws Exception {
+        // set up
+        Azure azure = PowerMockito.mock(Azure.class);
+        String instanceId = AzureGeneralUtil.defineInstanceId(AzureTestUtils.RESOURCE_NAME);
+        NetworkInterface networkInterface = Mockito.mock(NetworkInterface.class);
+
+        PublicIPAddresses publicIPAddresses = Mockito.mock(PublicIPAddresses.class);
+        Mockito.when(azure.publicIPAddresses()).thenReturn(publicIPAddresses);
+
+        PublicIPAddress publicIPAddress = Mockito.mock(PublicIPAddress.class);
+        Mockito.when(publicIPAddresses.getByResourceGroup(Mockito.eq(this.resourceGroupName), Mockito.eq(instanceId)))
+                .thenReturn(publicIPAddress);
+
+        Creatable<NetworkSecurityGroup> creatable = Mockito.mock(Creatable.class);
+        PowerMockito.mockStatic(AzurePublicIPAddressSDK.class);
+        PowerMockito.doReturn(creatable).when(AzurePublicIPAddressSDK.class, "buildNetworkSecurityGroupCreatable",
+                Mockito.eq(azure), Mockito.eq(publicIPAddress));
+
+        Observable<NetworkInterface> observable = Mockito.mock(Observable.class);
+        PowerMockito.doReturn(observable).when(AzurePublicIPAddressSDK.class, "associateNetworkSecurityGroupAsync",
+                Mockito.eq(networkInterface), Mockito.eq(creatable));
+
+        Mockito.doNothing().when(this.operation).subscribeUpdateNetworkInterface(Mockito.eq(observable));
+
+        // exercise
+        this.operation.handleSecurityIssues(azure, instanceId, networkInterface);
+
+        // verify
+        Mockito.verify(azure, Mockito.times(TestUtils.RUN_ONCE)).publicIPAddresses();
+        Mockito.verify(publicIPAddresses, Mockito.times(TestUtils.RUN_ONCE))
+                .getByResourceGroup(Mockito.eq(this.resourceGroupName), Mockito.eq(instanceId));
+
+        PowerMockito.verifyStatic(AzurePublicIPAddressSDK.class, Mockito.times(TestUtils.RUN_ONCE));
+        AzurePublicIPAddressSDK.buildNetworkSecurityGroupCreatable(Mockito.eq(azure), Mockito.eq(publicIPAddress));
+
+        PowerMockito.verifyStatic(AzurePublicIPAddressSDK.class, Mockito.times(TestUtils.RUN_ONCE));
+        AzurePublicIPAddressSDK.associateNetworkSecurityGroupAsync(Mockito.eq(networkInterface), Mockito.eq(creatable));
+
+        Mockito.verify(this.operation, Mockito.times(TestUtils.RUN_ONCE))
+                .subscribeUpdateNetworkInterface(Mockito.eq(observable));
+    }
+
+    // test case: When calling the subscribeUpdateNetworkInterface method and the
+    // observable executes without any error, it must verify than returns the right
+    // logs.
+    @Test
+    public void testSubscribeUpdateNetworkInterfaceSuccessfully() {
+        // set up
+        Observable<NetworkInterface> observable = Observable.defer(() -> {
```

Comment:
Same as the previous issue.

Reply:
The "Observable<Indexable>" type of the createSimpleObservableSuccess method in AzureTestUtils is not compatible with "Observable<NetworkInterface>".